esphome/components/airthings_wave_plus/airthings_wave_plus.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "esphome/core/component.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/preferences.h"
+#include "esphome/core/log.h"
+#include "esphome/core/application.h"
+#include "esphome/core/version.h"
```

Comment:
```suggestion
#include "esphome/core/component.h"
#include "esphome/core/helpers.h"
#include "esphome/core/log.h"

#ifdef ARDUINO_ARCH_ESP32
```

- I don't think all those includes are used
- Adds ifdef check so linter doesn't activate because that one uses an ESP8266 toolchain

Reply:
I had to keep #include `"esphome/core/application.h"` for `sensor` to be defined.